pkg/conversion/deep_equal.go

```diff
@@ -181,6 +184,9 @@ func (e Equalities) deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool,
 		}
 		return true
 	case reflect.Slice:
+		if v1.Len() != v2.Len() {
```

Comment:
Last comment: to maintain diffability with the std library function this is copied from, please move this below the nil check. https://golang.org/src/reflect/deepequal.go#L72